pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -438,6 +438,9 @@ func (tc *TidbCluster) TiDBStsDesiredOrdinals(excludeFailover bool) sets.Int32 {
 }
 
 func (tc *TidbCluster) PDIsAvailable() bool {
+	if tc.Spec.PD == nil {
+		return false
```

Comment:
We may have to return `true` here, or this may block the upgrade of other components.